Added build documentation. (#1)
eProsima/xtypes: README.md

```c++
//create the DynamicData accordingly with the recently created "Outer" DynamicType
DynamicData data(outer);

//write value

> data["b"]["a"].value(42);

Reviewer: So we decided to remove the functionality .value to do assignation? or we are just removing it from the documentation?

Author: Only from this part of the documentation (the first lines), to keep it simpler only show at this moment the *assignment* way.

Reviewer: it makes sense. agreed

- **No external dependency**: *eProsima xtypes*'s only dependencies are from *std*.
- **Easy to use**: Compresive API and intuitive concepts.

## Build

> *eprosima xtypes* is header only library, you can simply copy files located in the include folder into your project and include them.

Reviewer: *eprosima xtypes* is header only library: you can simply copy the files located in the include folder into your project and include them.

> *eprosima xtypes* is header only library, you can simply copy files located in the include folder into your project and include them.

Reviewer: ```suggestion
*eprosima xtypes* is a header only library: in order to use it you simply have to copy the files located in the include folder into your project and include them.
```

For a better management and version control, we recommend to make use of the **CMake** project that xtypes offers.

> Although there is no library generated by from *xtypes*, you can link with its target as following to enable your project to include the *xtypes* headers:

Reviewer: Although there is no library generated by *xtypes*, you can link with its target as following to enable the inclusion of *xtypes* headers:

Author: Could you suggest it the change please?

> Although there is no library generated by from *xtypes*, you can link with its target as following to enable your project to include the *xtypes* headers:

Reviewer: ```suggestion
Although there is no library generated by *xtypes*, you can do the linking with its target as following. This will enable the inclusion of *xtypes* headers:
```

```
target_link_libraries(${PROJECT_NAME} xtypes)
```

### Examples

> To compile the examples located in `examples/` folder, enable the `XTYPES_BUILD_EXAMPLES` cmake flag.

Reviewer: To compile the examples located in the `examples/` folder, enable the `XTYPES_BUILD_EXAMPLES` cmake flag.

> To compile the examples located in `examples/` folder, enable the `XTYPES_BUILD_EXAMPLES` cmake flag.

Reviewer: ```suggestion
To compile the examples located in the `examples/` folder, enable the `XTYPES_BUILD_EXAMPLES` cmake flag.
```
